integration_test/gce/gce_testing.go

```diff
@@ -1215,6 +1214,20 @@ func StartInstance(ctx context.Context, logger *log.Logger, vm *VM) error {
 	return waitForStart(ctx, logger, vm)
 }
 
+// RestartInstance stops and starts the instance.
+// It also waits for the instance to be reachable over ssh post-restart.
+func RestartInstance(ctx context.Context, logger *logging.DirectoryLogger, vm *VM) error {
+	fileLogger := logger.ToFile("VM_Restart.txt")
+
+	if err := StopInstance(ctx, fileLogger, vm); err != nil {
+		return fmt.Errorf("Failed to stop instance: %w", err)
+	}
+
+	time.Sleep(30 * time.Second)
```

Comment:
Nit: `30 * time.Second` should also have its own `vmSomethingBackoffDuration`

Reply:
Good callout, this sleep actually shouldn't even be necessary, I'll just remove it altogether.